.github/workflows/functional-tests.yml

```
@@ -0,0 +1,53 @@
+---
+name: functional-tests
+
+on:
+  workflow_dispatch: ~
+  schedule:
+    - cron: '0 3 * * 1-5'
+
+permissions:
+  contents: read
+  id-token: write
+
+jobs:
+  run:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        environment:
+          - 'qa'
+          - 'pro'
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'functionaltests/go.mod'
+
+      - uses: elastic/oblt-actions/google/auth@v1
+
+      - uses: google-github-actions/get-secretmanager-secrets@e5bb06c2ca53b244f978d33348d18317a7f263ce # v2.2.2
+        with:
+          export_to_environment: true
+          secrets: |-
+            EC_API_KEY:elastic-observability/elastic-cloud-observability-team-${{ matrix.environment }}-api-key
+
+      - run: cd functionaltests && go test -v -timeout=20m -target "${{ matrix.environment }}" ./
```

Let's honour the variable `CREATED_DATE`:

```suggestion
      - run: |
          export TF_VAR_CREATED_DATE=$(date +%s)
          cd functionaltests && go test -v -timeout=20m -target "${{ matrix.environment }}" ./
```